src/utils/jws_utils/src/jws_utils.c

```diff
@@ -396,6 +517,71 @@ JWSResult VerifyJWSWithSJWK(const char* jws)
     return result;
 }
 
+/**
+ * @brief Whether signing key in SJWK (header of JWS) is no longer trusted as per latest root key package on file.
+ *
+ * @param sjwkJsonStr The SJWK json string that is base64 URL decoded header section of the JWS.
+ * @param disabledHashOfPubKeysList The list of sha256 hashes of public keys of signing keys that are disabled.
+ * @return JWSResult Returns JWSResult_Success if not on Disallowed; JWSResult_DisallowedSigningKey if signing key was on Disallowed, or other failure JWSResult if failed whilst determining.
+ */
+JWSResult IsSigningKeyDisallowed(const char* sjwkJsonStr, VECTOR_HANDLE disabledHashOfPubKeysList)
+{
+    JWSResult result = JWSResult_Failed;
+    CONSTBUFFER_HANDLE pubkey = NULL;
+    CONSTBUFFER_HANDLE sha256HashPubKey = NULL;
+
+    char* N = GetStringValueFromJSON(sjwkJsonStr, "n");
+    char* e = GetStringValueFromJSON(sjwkJsonStr, "e");
+    if (IsNullOrEmpty(N) || IsNullOrEmpty(e) || strcmp(e, "AQAB") != 0) // AQAB is 65337 or 0x010001, the ubiquitous RSA exponent.
+    {
+        result = JWSResult_InvalidSJWKPayload;
+        goto done;
+    }
+
+    pubkey = CryptoUtils_GeneratePublicKey(N, e);
+    if(pubkey == NULL)
+    {
+        result = JWSResult_FailGenPubKey;
+        goto done;
+    }
+
+    sha256HashPubKey = CryptoUtils_CreateSha256Hash(pubkey);
+    if(pubkey == NULL)
+    {
+        result = JWSResult_HashPubKeyFailed;
+        goto done;
+    }
+
+    // See if the hash of public key is on the Disallowed.
+    for(int i=0; i<VECTOR_size(disabledHashOfPubKeysList); ++i)
```

Comment:
nit: run this PR through clang-format.

Reply:
fixed with https://github.com/Azure/iot-hub-device-update/pull/489/commits/69c022e7af65368415a106ade88fcc87ea0a7015 on upstream topic branch